### PR TITLE
Fix/oo#438 producer nickname duplication

### PR DESCRIPTION
**Pull request by mcmpp**

### CODE REVIEW

File: app/scripts/produzenten/detail/produzentendetail.controller.js [hunk 1]
```diff
@@ -54,14 +54,15 @@ angular.module('openolitor-admin')
 
       $scope.save = function() {
         var listOfNicknames = lodash.map($scope.produzenten, 'kurzzeichen');
+        var normalizedListOfNicknames = listOfNicknames.map(nickname => nickname.normalize('NFD').replace(/[\u0300-\u036f]/g, ""));
         //new producer
         //normalization in order to remove the accents. https://stackoverflow.com/questions/990904/remove-accents-diacritics-in-a-string-in-javascript/37511463#37511463
-        if (isNew && lodash.filter(listOfNicknames, function(n){ return n === $scope.produzent.kurzzeichen.normalize('NFD').replace(/[\u0300-\u036f]/g, "")}).length > 0){
+        if (isNew && lodash.filter(normalizedListOfNicknames, function(n){ return n === $scope.produzent.kurzzeichen.normalize('NFD').replace(/[\u0300-\u036f]/g, "")}).length > 0){
```
Comment (robstoll) [inline app/scripts/produzenten/detail/produzentendetail.controller.js]: I would move the normalizing part into an own function and reuse it for nickname and produzent.kurzzeichen

File: package.json [hunk 2]
```diff
@@ -29,7 +29,7 @@
     "grunt-contrib-imagemin": "^3.1.0",
     "grunt-contrib-jshint": "^1.1.0",
     "grunt-contrib-symlink": "^1.0.0",
-    "grunt-contrib-uglify": "^3.0.1",
+    "grunt-contrib-uglify": "^5.0.0",
```
Comment (robstoll) [inline package.json:32]: as this is unrelated to PR, on purpose?

File: app/scripts/produzenten/detail/produzentendetail.controller.js (hunk 1 above)
Comment (mcmpp) [reply app/scripts/produzenten/detail/produzentendetail.controller.js]: done

File: package.json (hunk 2 above)
Comment (mcmpp) [reply package.json:32]: the version of uglify that we are using does not support maps. I am using one and it would make the build fail on the uglify step. I do not see any drawback from upgrading it

**Review by robstoll — APPROVED** [review]
